src/workerd/io/io-gate.h

```diff
@@ -99,12 +101,17 @@ class InputGate {
 
     kj::Maybe<kj::Own<CriticalSection>> cs;
 
-    Lock(InputGate& gate);
+    SpanBuilder lockSpan;
+
+    Lock(InputGate& gate, SpanParent parentSpan);
     friend class InputGate;
   };
 
   // Wait until there are no `Lock`s, then create a new one and return it.
-  kj::Promise<Lock> wait();
+  // If parentSpan is provided, child spans will be created to track:
+  // - Time spent waiting for the lock (if waiting is required)
+  // - Time spent holding the lock
+  kj::Promise<Lock> wait(SpanParent parentSpan);
```

Comment:
nit: it would be better to have an extra line of comment space between the first line of documentation and the parentSpan documentation.

```
  // Wait until there are no `Lock`s, then create a new one and return it.
  //
  // If parentSpan is provided, child spans will be created to track:
  // - Time spent waiting for the lock (if waiting is required)
  // - Time spent holding the lock
  kj::Promise<Lock> wait(SpanParent parentSpan);
```